in_toto/models/metadata.py

```diff
@@ -82,6 +82,7 @@ def dump(self, filename):
       None.
 
     """
+    filename = path + filename
```

Comment:
I'm leaving this comment here, but it may apply elsewhere if we decide to perform this operation elsewhere and pass in a full path to `dump`. I suggest using `os.path.join` to generate the full path, as this also uses the correct separators, as well as adding a trailing separator that may have been missed.

We should also check if the value passed for `metadata_directory` is a directory, and throw an error if not. I also recommend catching any permission based errors - what if the user doesn't have the permission to write to the specified directory?